src/util/loadDatasetGraph.ts

```diff
@@ -85,11 +99,77 @@ function createDatasetGraph(rawGraph: any): DatasetGraph {
     for (const term of ontology.values()) {
       term.in_use = !!term.n_cells || [...term.descendants].some((id) => ontology.get(id)!.n_cells > 0);
     }
+    const rootKeys = [...ontology].filter(([_,v])=>v.parents?.length===0).map(([k,_])=>k);    
+    rootKeys.forEach((key)=>{
+      const rootTerm = ontology.get(key);  
+      if (rootTerm) {
+        calcDepthBFS(ontology, rootTerm, datasetGraph.depthMaps[prefix])
+        calcHeightDFS(ontology, rootTerm, datasetGraph.heightMaps[prefix])        
+      }
+    })    
   }
 
   return datasetGraph;
 }
 
+/**
+ * Calculate depth of a node in the ontology
+ *
+ * @param ontology
+ * @param term
+ * @param depthMap
+ */
+ function calcDepthBFS(ontology: Ontology, term: OntologyTerm, depthMap: Map<string,number>): void {
+  const stack = [...term.children];
+  const depthStack: number[] = new Array(stack.length);
+  depthStack.fill(1);
+  depthMap.set(term.id,0);
+  if (term.depth===-1) term.depth=0;
+  while ( stack.length > 0 ) {
```

Comment:
lint/formatting - please run prettier with the project config before committing....